lib/mix/lib/mix/tasks/test.ex

```diff
@@ -175,6 +175,24 @@ defmodule Mix.Tasks.Test do
   def run(args) do
     {opts, files} = OptionParser.parse!(args, strict: @switches)
 
+    if opts[:listen_on_stdin] do
+      try do
+        Process.flag(:trap_exit, true)
```

Comment:
Here is the code we have on the CLI but I am hoping it won't have to be this complex: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/kernel/cli.ex#L70